### PR TITLE
Enhance/shared babel

### PR DESCRIPTION
**Pull request by u9520107**
1. Create an eslint-settings package for shared eslint configuration and dependencies.
2. Create a babel-settings package for shared babel configuration and dependencies.
3. Use babel-preset-env and explicitly list feature that's not yet in the spec.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,8 +1,15 @@
...
+    "env"
+  ],
   "plugins": [
+    "transform-object-rest-spread",
+    "transform-decorators-legacy",
     "transform-runtime",
-    "transform-decorators-legacy"
+    "transform-function-bind", // stage 0
+    "transform-export-extensions", // stage 1
+    "transform-class-properties" // stage 2
```
Comment (ele828) [inline .babelrc]: Would you also consider extending babelrc from babel-settings?

File: packages/monorepo-tools/package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
   "scripts": {},
   "dependencies": {
     "@yarnpkg/lockfile": "^1.0.0",
+    "babel-settings": "0.0.1",
```
Comment (ele828) [inline packages/monorepo-tools/package.json]: devDependencies?

File: package.json [hunk 3]
```diff
@@ -20,13 +20,11 @@
     "analyze-packages": "babel-node scripts/analyzePackages.js",
     "list-file-matchers": "babel-node scripts/listFileMatchers.js"
   },
+  "dependencies": {
+    "babel-settings": "0.0.1"
```
Comment (ele828) [inline package.json]: devDependencies?

File: packages/monorepo-tools/package.json (hunk 2 above)
Comment (u9520107) [reply packages/monorepo-tools/package.json]: babel-settings is more complicated, babel-runtime is a runtime dependency rather than a dev dependency.

Comment (u9520107) [reply packages/monorepo-tools/package.json]: hmm... good point though.. let me reconsider this.

File: package.json [hunk 4]
```diff
@@ -21,12 +21,8 @@
     "list-file-matchers": "babel-node scripts/listFileMatchers.js"
   },
   "devDependencies": {
-    "babel-eslint": "^8.2.3",
-    "eslint": "^4.12.0",
-    "eslint-config-airbnb": "^16.1.0",
-    "eslint-plugin-import": "^2.8.0",
-    "eslint-plugin-jsx-a11y": "^6.0.2",
-    "eslint-plugin-react": "^7.5.1",
+    "eslint-settings": "0.0.1",
```
[the remark this answers is not in the document]
Comment (u9520107) [reply package.json]: Don't think so. I've done a ```git clean -xdf``` to clear out all the files to do a clean install. Plus all the CI tests passed, so the installation should be fine.

**Pull request merged**